DigitalLearningSolutions.Data/Services/ActionPlanService.cs

```diff
@@ -185,5 +184,24 @@ public void RemoveActionPlanResource(int learningLogItemId, int delegateId)
 
             return actionPlanResource.LoggedById == delegateId;
         }
+
+        private async Task<IEnumerable<ActionPlanResource>> MapLearningLogItemsToActionPlanResources(
+            IList<LearningLogItem> learningLogItems
+        )
+        {
+            if (!learningLogItems.Any())
+            {
+                return new List<ActionPlanResource>();
+            }
+
+            var resourceIds = learningLogItems.Select(i => i.LearningHubResourceReferenceId!.Value);
+            var bulkResponse = await learningHubApiClient.GetBulkResourcesByReferenceIds(resourceIds);
+            return bulkResponse.ResourceReferences.Select(
+                resource => new ActionPlanResource(
+                    learningLogItems.Single(i => i.LearningHubResourceReferenceId!.Value == resource.RefId),
+                    resource
+                )

```

Comment:
Is this single not now dangerous since we can have the same resource completed mutliple times?

Reply:
Good catch. I've refactored this to now be able to deal with multiple learning log items, and implemented a unit test for this case. 

I've kept the logic in the common code, so multiple Current items can also be displayed. I think that is acceptable, as the responsibility of keeping only a single Current item should be tackled when adding items to the plan.